nixos/modules/services/databases/postgresql.nix

```diff
@@ -631,6 +744,48 @@ in
         unitConfig.RequiresMountsFor = "${cfg.dataDir}";
       };
 
+      system.activationScripts.detect-previous-postgresql-installation = lib.mkIf (cfg.upgrade.enable && cfg.upgrade.enablePreviousInstallationAutodetection) {
+        deps = [ "etc" ];
+        text = ''
+          previousPgDetect() {
+            echo "Trying to detect prevoius PostgreSQL installation, because 'current' symlink does not exist in 'config.services.postgresql.databaseDir'."
+
+            if [[ ! -x /run/current-system/sw/bin/systemctl ]]; then
+              echo "systemctl binary is missing or is not executable. This is ok on fresh deployments. Not running previous PostgreSQL installation detection."
+              return
+            fi
```

Comment:
I've added that after a failure in a test, which runs from a clean state. But I've fixed some things later, so might not be needed afetr all. Will check

Reply:
If you remove it, you'll get this:
```
vm-test-run-postgresql-upgrade> machine # running activation script...
vm-test-run-postgresql-upgrade> machine # setting up /etc...
vm-test-run-postgresql-upgrade> machine # Trying to detect prevoius PostgreSQL installation, because 'current' symlink does not exist in 'config.services.postgresql.databaseDir'.
vm-test-run-postgresql-upgrade> machine # /nix/store/s2wylhnw3c6ayyjagdmjgf9afh9sazdz-nixos-system-machine-test/activate: line 75: /run/current-system/sw/bin/systemctl: No such file or directory
vm-test-run-postgresql-upgrade> machine # Cannot load old PostgreSQL Environment from systemctl.
```
but the `|| true` on the next line will catch it.

So this only provides nicer message currently